File: src/main/java/com/jaredsburrows/license/internal/pom/License.java
package com.jaredsburrows.license.internal.pom;

import java.util.Objects;

/**
 * Represents license information in a POM file.
 */
public final class License {
  private String name;
  private String url;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getUrl() {
    return url;
  }

  public void setUrl(String url) {
    this.url = url;
  }

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof License)) return false;
    License license = (License) o;
    return Objects.equals(url, license.url);
  }

  @Override public int hashCode() {
    return Objects.hash(url);
  }
}

File: src/main/kotlin/com/jaredsburrows/license/internal/pom/License.kt
package com.jaredsburrows.license.internal.pom

/**
 * Represents license information in a
 * [POM](https://maven.apache.org/guides/introduction/introduction-to-the-pom.html) file.
 *
 * @property name name of the [License] in the POM.
 * @property url url of the [License] in the POM.
 */
data class License(
  var name: String = "",
  var url: String = ""
) {
  override fun equals(other: Any?): Boolean {
    return when {
      this === other -> true
      other !is License -> false
      url != other.url -> false
      else -> true
    }
  }

  override fun hashCode(): Int = url.hashCode()
}
